lib/graphql/schema/rescue_middleware.rb

```diff
@@ -41,13 +41,14 @@ def call(*args)
       private
 
       def attempt_rescue(err)
-        handler = rescue_table[err.class]
-        if handler
-          message = handler.call(err)
-          GraphQL::ExecutionError.new(message)
-        else
-          raise(err)
-        end
+        rescue_table.each { |klass, handler|
+          if klass.is_a?(Class) && err.is_a?(klass) && handler
```

Comment:
I understand one of the conditions: `err.is_a?(klass)`, but what about `klass.is_a?(Class)` and checking for `handler`? Are those conditions required, or can we remove them? 

Or, if these checks are necessary, do you know if we have tests for that case?

Reply:
I opted to go for the safest scenario possible: 

since there is no validation when populating the `rescue_table` `klass.is_a?(Class)` ensures that we only run `err.is_a?(klass)` if in fact `klass` is a `Class`, otherwise we'd get a `TypeError` .

the `handler` presence check was for the similar reasons, nothing forbids us from calling `rescue_from(Error)` without the block, which would lead to `handler=nil` right?

Reply:
let me know if this makes sense to you or if perhaps we should take a more conservative approach and `raise` an error when `error_classes` provided to `rescue_from` are not a `class` in https://github.com/rmosolgo/graphql-ruby/pull/1393/files/651a12bdae4db9355eefbff09eafe0bd12890309#diff-b8d49293896b9144b1b73bd7906d7513R22